Sources/Common/Core/DataArray/index.d.ts

```diff
@@ -83,6 +83,19 @@ export interface vtkDataArray extends vtkObject {
    */
   setRange(rangeValue: vtkRange, componentIndex: number): Range;
 
+  /**
+   * Returns an array of the ranges for each component of the DataArray.
+   * Defaults to computing all the ranges if they aren't already computed.
+   *
+   * Passing `getRanges(false)` will return  a clone of the ranges that have
+   * already been computed. This is useful when you want to avoid recomputing
+   * the ranges, which can be expensive.
```

Comment:
Please add a note about the last element in the returned ranges being the min,max magnitude of the entire dataset. This can be the same as your comment in publicAPI.getRanges().